lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -24,33 +24,26 @@ Import-Module JujuHooks
 Import-Module OpenStackCommon
 
 function Get-Vcredist {
-    BEGIN {
+    Write-JujuWarning "Getting vcredist."
+    $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
+    if(!$vcredistUrl) {
         $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
```

Comment:
Totally agree with you on this. But we have a retry for `Get-JujuResource` (and everyone should have that since it downloads stuff from the state machine) and I was picturing the change, which looks ugly as well. :)

You cannot have an external retry to the `Get-JujuResource`. Imagine the scenario when the resources are not there, all the retries will just waste time until you get into the catch block.

This can be solved if you add internal retry to `Get-JujuResource` function.